dashboard/app/models/script.rb

```diff
@@ -888,66 +888,78 @@ def has_older_version_progress?(user)
-        name = File.basename(script, '.script')
-        base_name = Script.base_name(name)
-        name = "#{base_name}-#{new_suffix}" if new_suffix
-        script_data, i18n =
-          begin
-            ScriptDSL.parse_file(script, name)
-          rescue => e
-            raise e, "Error parsing script file #{script}: #{e}"
-          end
-
-        lesson_groups = script_data[:lesson_groups]
-        custom_i18n.deep_merge!(i18n)
-        # TODO: below is duplicated in update_text. and maybe can be refactored to pass script_data?
-        scripts_to_add << [{
-          id: script_data[:id],
-          name: name,
-          hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
-          login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
-          wrapup_video: script_data[:wrapup_video],
-          new_name: script_data[:new_name],
-          family_name: script_data[:family_name],
-          properties: Script.build_property_hash(script_data).merge(new_properties)
-        }, lesson_groups]
-      end
+    scripts_to_add = []
+
+    custom_i18n = {}
+    # Load custom scripts from Script DSL format
+    custom_files.map do |script|
+      name = File.basename(script, '.script')
+      base_name = Script.base_name(name)
+      name = "#{base_name}-#{new_suffix}" if new_suffix
+      script_data, i18n =
+        begin
+          ScriptDSL.parse_file(script, name)
+        rescue => e
+          raise e, "Error parsing script file #{script}: #{e}"
+        end
 
-      # Stable sort by ID then add each script, ensuring scripts with no ID end up at the end
-      added_scripts = scripts_to_add.sort_by.with_index {|args, idx| [args[0][:id] || Float::INFINITY, idx]}.map do |options, raw_lesson_groups|
-        add_script(options, raw_lesson_groups, new_suffix: new_suffix, editor_experiment: new_properties[:editor_experiment])
-      rescue => e
-        raise e, "Error adding script named '#{options[:name]}': #{e}", e.backtrace
-      end
-      [added_scripts, custom_i18n]
+      lesson_groups = script_data[:lesson_groups]
+      custom_i18n.deep_merge!(i18n)
+      # TODO: below is duplicated in update_text. and maybe can be refactored to pass script_data?
+      scripts_to_add << [{
+        id: script_data[:id],
+        name: name,
+        hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
+        login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
+        wrapup_video: script_data[:wrapup_video],
+        new_name: script_data[:new_name],
+        family_name: script_data[:family_name],
+        properties: Script.build_property_hash(script_data).merge(new_properties)
+      }, lesson_groups]
+    end
+
+    # Print progress for dev environments and adhocs
+    debug = [:adhoc, :development].include?(rack_env)
+    if debug
+      $stdout.sync = true
```

Comment:
what does this do? just wondering if it needs to be undone after, so that it doesn't affect the rest of `rake seed:all`.

Reply:
I just copied it from here: https://stackoverflow.com/questions/5080644/how-can-i-use-puts-to-the-console-without-a-line-break-in-ruby-on-rails/5080779#5080779

Sounds like I can also call `$stdout.flush` after each print for the same effect. Should I change to doing that? I was hoping someone had opinions on best practice here.

Reply:
I see.. I was worried this might make all the later output appear on one line, but it looks like it just makes the output appear more promptly.

the comments on that answer seem to suggest that `sync` might make the rest of your output slightly slower. If you are looking for opinions, `flush` seems slightly better to me since it will not have any side effects on the rest of your program. the rest of the seed process usually doesn't have much output... except when you are performance debugging... which could be a really sad time to have a hidden performance hit due to the `sync` setting 😂

Reply:
Yea, I did find that the SQL query printing did impact the benchmarks, since there are so many lines. But I think printing once per script should have no impact either way. Would be funny/sad though. Changed to using flush.